src/wallet/mod.rs

```diff
@@ -11,196 +11,343 @@ use pbkdf2;
 use sha2::Sha256;
 use shamirsecretsharing::hazmat::{combine_keyshares, create_keyshares};
 use sodiumoxide::randombytes;
-use std::io::{self, Cursor};
+use std::{
+    boxed::Box,
+    fmt,
+    io::{self, Cursor},
+};
 
 pub type Salt = [u8; 8];
 pub type Tag = [u8; 16];
 pub type IV = [u8; 12];
 pub type AESKey = [u8; 32];
 pub type SSSKey = [u8; 32];
 
+#[derive(Clone)]
+pub struct KeyShare(pub(crate) [u8; 33]);
+
 type HmacSha256 = Hmac<Sha256>;
 
-mod basic_wallet;
-pub mod basic {
-    pub use crate::wallet::basic_wallet::*;
-}
-mod sharded_wallet;
-pub mod sharded {
-    pub use crate::wallet::sharded_wallet::*;
+const WALLET_KIND_BASIC: u16 = 0x0001;
+const WALLET_KIND_SHARDED: u16 = 0x0101;
+const WALLET_DEFAULT_ITERATIONS: u32 = 1_000_000;
+
+impl Default for KeyShare {
+    fn default() -> Self {
+        KeyShare([0; 33])
+    }
 }
 
-pub enum Wallet {
-    Basic(basic::Wallet),
-    Sharded(sharded::Wallet),
+impl KeyShare {
+    pub fn to_vec(&self) -> Vec<u8> {
```

Comment:
I feel like there's a lot data copying going on here. Would make sense to change this to `as_slice()` and/or  `impl AsRef<[u8]> for KeyShare`? 

 I know it probably doesn't matter performance-wise, but if this ever winds up as a library instead of a monolithic binary, the API might be a bit unwieldy.

Reply:
yes? maybe?